api/client/modelmanager/legacy.go

```diff
@@ -0,0 +1,197 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package modelmanager
```

Comment:
Can you put a paragraph at the top of this regarding the change instituted that required legacy compatibility?